web/src/store/index.ts

```diff
@@ -94,6 +106,45 @@ const {
           const activeDataset = tileSourceDatasets.length > 0 ? tileSourceDatasets[0] : null;
           commit.setActiveDataset(activeDataset);
 
+          const embeddings: DatasetEmbedding[] = (await store.state.axiosInstance.get(`/investigations/${investigationId}/embeddings`)).data;
+          commit.setDatasetEmbeddings(embeddings);
+
+          const metadataPromises: Promise<{ datasetId: string; result: AxiosResponse }>[] = [];
+          tileSourceDatasets.forEach((dataset) => {
+            const promise = store.state.axiosInstance?.get(`/datasets/${dataset.id}/tiles/metadata`).then((result) => ({
+              datasetId: dataset.id,
+              result,
+            }));
+            if (promise) {
+              metadataPromises.push(promise);
+            }
+          });
+          embeddings.forEach((embedding) => {
```

Comment:
So with your recent changes, we're retrieving and storing the metadata for all `Datasets` that are:

1. Related to the investigation through the `Investigation.datasets` relation an of type `tile_source`
2. Related to the investigation by being either the `parent` or `child` property of an `Embedding`

Since a `Dataset` can be included in both (1) and (2), I think there's some redundancy with the current approach. I'm fine to leave it as is for now, since it works, but making a set of `Dataset` ids and then building the promise array from that might be a cleaner option.

I imagine that all Datasets in (2) should also be in (1), but I don't think our DB enforces that, and its possible that my interpretation of our data model is slightly off. But that's a separate discussion.

Reply:
> Since a `Dataset` can be included in both (1) and (2), I think there's some redundancy with the current approach. I'm fine to leave it as is for now, since it works, but making a set of `Dataset` ids and then building the promise array from that might be a cleaner option.

Totally agree, but, I think that array of all datasets is used for the dropdown selector in the original view. I was afraid adding the `Dataset` IDs related to the embeddings to that array would have unintended consequences.

Reply:
I've opened #110 to capture my perspective on investigation/dataset relationships. Your approach here is valid.